cloud/docker_client.go

```diff
@@ -96,6 +163,11 @@ func (c *dockerClientImpl) CreateContainer(ctx context.Context, h *host.Host, na
 		return errors.Wrap(err, "Failed to generate docker client")
 	}
 
+	settings, err := evergreen.GetConfig()
+	if err != nil {
+		return errors.Wrap(err, "Error retrieving Evergreen settings")
+	}
```

Comment:
in generally it's bad news (for testing) if you're calling GetConfig here/this way because it's a little "global magic" 

evergreen.Settings objects should be passed into the client at creation time.